From 5549ea64799784308cc03313a86dea3de56d48ce Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Tue, 10 Mar 2020 11:35:07 -0500
Subject: [PATCH] ASoC: rt5682: fix unmet dependencies
Git-commit: 5549ea64799784308cc03313a86dea3de56d48ce
Patch-mainline: v5.7-rc1
References: jsc#SLE-16518

The rt5682 code can be used in I2C or SoundWire mode. When I2C is not
selected, we have the following issue:

Warning: unmet direct dependencies detected for SND_SOC_RT5682  Depends on [n]: SOUND [=m] && !UML && SND [=m] && SND_SOC [=m] &&  I2C [=n]  Selected by [m]:  - SND_SOC_RT5682_SDW [=m] && SOUND [=m] && !UML && SND [=m] &&    SND_SOC [=m] && SOUNDWIRE [=m]

Fix by adding SOUNDWIRE as a dependency.

Fixes: 03f6fc6de9192f ('ASoC: rt5682: Add the soundwire support')
Reported-by: kbuild test robot <lkp@intel.com>
Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Cc: Oder Chiou <oder_chiou@realtek.com>
Link: https://lore.kernel.org/r/20200310163509.14466-2-pierre-louis.bossart@linux.intel.com
Signed-off-by: Mark Brown <broonie@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 sound/soc/codecs/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/codecs/Kconfig b/sound/soc/codecs/Kconfig
index 6aee70ed43df..78be69e9b618 100644
--- a/sound/soc/codecs/Kconfig
+++ b/sound/soc/codecs/Kconfig
@@ -1135,7 +1135,7 @@ config SND_SOC_RT5677_SPI
 
 config SND_SOC_RT5682
 	tristate
-	depends on I2C
+	depends on I2C || SOUNDWIRE
 
 config SND_SOC_RT5682_SDW
 	tristate "Realtek RT5682 Codec - SDW"
-- 
2.16.4

